[TelemetryHttpModule] Update `OpenTelemetry.Api` reference and changelog

vishweshbankwar opened the pull request:
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #

Kielek commented:
Probably I have missed something, but what is a reason for the release? CHANGELOG.md is empty.

BTW consider to update reference to OTel.Api to 1.3.1 as 1.3.0 is unlisted.

vishweshbankwar commented:
> Probably I have missed something, but what is a reason for the release? CHANGELOG.md is empty.
> 
> BTW consider to update reference to OTel.Api to 1.3.1 as 1.3.0 is unlisted.

Good catch on Api reference. I have updated it. We keep the version for this one same as [OpenTelemetry.Instrumentation.AspNet](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNet). That was the reason for empty changelog before, I had updated the version for instrumentation lib but missed this one.
